onnxruntime/test/optimizer/compute_optimizer_test.cc

```diff
@@ -1831,6 +1831,72 @@ TEST(ComputeOptimizerTests, ReshapeElementwiseOps_NoPropagation1) {
   }
 }
 
+/*
+Test graph include multiple equivalent subgraphs as below.
+                           graph input [128, 4, 32] (int64_t)
+                            |               |
+                                          Cast
+                                            |
+                                         Reshape
```

Comment:
Add the shape initializer [-1, 128] here?

Reply:
fixed.